### ISSUE TITLE
Add a `configs.customize()` factory beside `configs.recommended`

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#746]
### What is the problem you're trying to solve?

`configs.recommended` (#745) decides everything itself, and anyone who disagrees overrides the rule in `rules`; a single-syntax project needs one line under the right name. Two cases are left short. A project with several syntaxes has to change one value under every name of the rule: `@stylistic/indentation`, `@stylistic/scss/indentation`, `@stylistic/less/indentation`. And `severity` and `disableFix` for all the stylistic rules at once go into the secondary options of each of them, which Stylelint's `defaultSeverity` cannot express.

### What solution would you like to see?

A factory like `configs.customize()` in `@stylistic/eslint-plugin`: it changes a cell of the table before the table is projected onto the syntaxes, so every name gets the new value at once. `recommended` becomes `customize()` with every option at its default and keeps working as a subpath string.

```js
import stylistic from "@stylistic/stylelint-plugin"

export default {
	extends: [
		"stylelint-config-standard",
		stylistic.configs.customize({
			indent: 2,
			quotes: "single"
		}),
	],
}
```

**Options** are what actually varies between CSS projects: indentation, quotes, the number of consecutive empty lines, whether single-line blocks are allowed, line length, line breaks, `severity` and `disableFix` for all rules at once, and `syntaxes`, globs or `false` per namespace, with `scss` and `less` on by default and `styled` off, enabled with globs of your own. The defaults are the cells of `recommended`. Everything else the preset decides itself, as before.

**Guarantees.** The preset passes the contradicting-settings check under every option, every option changes exactly the rule it names, under every name, and `customize()` with no options equals `recommended`. An unknown key or a wrong value is ruled out.

**Blocker.** The factory returns an object, and it goes into `extends` as an object. The Stylelint runtime has accepted that since 15.9 (stylelint/stylelint#6998), but the `Config` type does not yet: under `@type {import("stylelint").Config}` such a config fails `tsc`. stylelint/stylelint#9525 fixes that; until it is released, the factory's docs would have to show a type cast, so this issue waits.
